app/models/credit.rb

```diff
@@ -5,4 +5,12 @@ class Credit < ActiveRecord::Base
 
   validates :participant_id, presence: true
   validates :credit_role_id, presence: true
+
+  before_save :setup_order
+
+  private
+
+  def setup_order
+    self.order ||= (post.credits.map(&:order).compact.max || 0) + 1
```

Comment:
Such as rails console or test.
However this code is unnecessary for production code.

I'll drop `Credit#setup_order`.